### PR TITLE
Cleanup volumesnapshorestore CRD

### PR DESCRIPTION
**Pull request by ram-infrac**
**What type of PR is this?**
>cleanup

**What this PR does / why we need it**:

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:

```release-note

```

**Does this change need to be cherry-picked to a release branch?**:
yes, 2.3.0

Integration-Test 

```
=== RUN   TestSnapshotRestore/cloudSnapshotRestoreTest
=== RUN   TestSnapshotRestore/groupCloudSnapshotRestoreTest
--- PASS: TestSnapshotRestore (181.00s)
    --- PASS: TestSnapshotRestore/simpleSnapshotRestoreTest (70.77s)
    --- PASS: TestSnapshotRestore/groupSnapshotRestoreTest (110.23s)
    --- SKIP: TestSnapshotRestore/cloudSnapshotRestoreTest (0.00s)
        snapshot_restore_test.go:74: Skipping CloudSnapRestore integration tests
    --- PASS: TestSnapshotRestore/groupCloudSnapshotRestoreTest (0.00s)
=== RUN   TestSnapshot
--- PASS: TestSnapshot (0.00s)
PASS

=== Skipped
=== SKIP: . TestSnapshotRestore/cloudSnapshotRestoreTest (0.00s)
    --- SKIP: TestSnapshotRestore/cloudSnapshotRestoreTest (0.00s)
        snapshot_restore_test.go:74: Skipping CloudSnapRestore integration tests


DONE 13 tests, 1 skipped in 230.801s
Tests passed
```

### CODE REVIEW

**Comment by disrani-px** [comment]
@ram-infrac Can you share the output of `kubectl describe volumesnapshotrestore` for a completed restore

**Comment by ram-infrac** [comment]
@disrani-px here,
```
Name:         mysql-snap-inrestore
Namespace:    default
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"stork.libopenstorage.org/v1alpha1","kind":"VolumeSnapshotRestore","metadata":{"annotations":{},"name":"mysql-snap-inrestore...
API Version:  stork.libopenstorage.org/v1alpha1
Kind:         VolumeSnapshotRestore
Metadata:
  Creation Timestamp:  2019-09-19T19:31:39Z
  Generation:          5
  Resource Version:    7354
  Self Link:           /apis/stork.libopenstorage.org/v1alpha1/namespaces/default/volumesnapshotrestores/mysql-snap-inrestore
  UID:                 1a21a89f-db14-11e9-8641-000c295d6364
Spec:
  Group Snapshot:    false
  Source Name:       mysql-snapshot
  Source Namespace:  default
Status:
  Pvcs:
    Mysql - Data:  default
  Restore Volumes:
    pvc-e0ec8ac8-db11-11e9-8641-000c295d6364:  k8s-volume-snapshot-f4146de7-db13-11e9-9574-6282cf9fd501
  Status:                                      Successful
  Volumes:
    pvc-e0ec8ac8-db11-11e9-8641-000c295d6364:
      Reason:    restore is successful
      Snapshot:  891264757075180373
      Status:    Successful
      Volume:    pvc-e0ec8ac8-db11-11e9-8641-000c295d6364
Events:          <none>
```

File: test/integration_test/common_test.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func setup() error {
 
 	provisioner := os.Getenv(storageProvisioner)
 	logrus.Infof("Using provisioner: %s", provisioner)
-	if err = volumeDriver.Init(schedulerDriverName, nodeDriverName, provisioner); err != nil {
+	if err = volumeDriver.Init(schedulerDriverName, nodeDriverName, "", provisioner); err != nil {
```
Comment (ram-infrac) [inline test/integration_test/common_test.go]: vol driver now accept `token` while initialising driver

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -1264,67 +1267,53 @@ func (p *portworx) CompleteVolumeSnapshotRestore(snapRestore *stork_crd.VolumeSn
...
+	restoreID := string(snapRestore.GetUID())
+	for _, vol := range snapRestore.Status.Volumes {
+		// Detect cloudsnap or local snapshot here
+		snapID, snapType, _, err := getSnapshotDetails(vol.Snapshot)
+		if err != nil {
+			log.VolumeSnapshotRestoreLog(snapRestore).Errorf("Invalid snapshot data %v", err)
+			return err
+		}
+
+		logrus.Infof("Restoring volume %v with local snapshot %v", vol.Volume, snapID)
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Use `log.VolumeSnapshotRestoreLog()`

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -1264,67 +1267,53 @@ func (p *portworx) CompleteVolumeSnapshotRestore(snapRestore *stork_crd.VolumeSn
...
+		if snapType == crdv1.PortworxSnapshotTypeCloud {
+			snapID = restoreNamePrefix + getUidforRestore(vol.Volume, restoreID)
 		}
 
 		err = wait.ExponentialBackoff(restoreAPICallBackoff, func() (bool, error) {
-			err = volDriver.Restore(volID, snapID)
+			err = volDriver.Restore(vol.Volume, snapID)
 			if err != nil {
-				logrus.Warnf("In-place restore failed for %v: %v", volID, err)
+				logrus.Warnf("In-place restore failed for %v: %v", vol.Volume, err)
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Use `log.VolumeSnapshotRestoreLog()`

File: drivers/volume/portworx/portworx.go [hunk 4]
```diff
@@ -1333,11 +1322,15 @@ func (p *portworx) pxSnapshotRestore(
...
 			logrus.Errorf("Unable to restore volume %v with ID %v, err %v",
-				volID, snapID, err)
+				vol.Volume, snapID, err)
+			vol.Reason = fmt.Sprintf("failed to do in-place restore, %v", err.Error())
+			vol.RestoreStatus = stork_crd.VolumeSnapshotRestoreStatusFailed
 			return err
 		}
-		logrus.Infof("Completed restore for volume %v with Snapshotshot %v", volID, snapID)
-		if isCloudSnap {
+		logrus.Infof("Completed restore for volume %v with Snapshotshot %v", vol.Volume, snapID)
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Use `log.VolumeSnapshotRestoreLog()`

File: drivers/volume/portworx/portworx.go [hunk 5]
```diff
@@ -1333,11 +1322,15 @@ func (p *portworx) pxSnapshotRestore(
...
+		vol.Reason = "restore is successful"
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: `Restore`

File: pkg/apis/stork/v1alpha1/snapshotrestore.go [hunk 6]
```diff
@@ -24,42 +23,40 @@ type VolumeSnapshotRestoreSpec struct {
...
 const (
 	// VolumeSnapshotRestoreStatusInitial is the initial state when snapshot restore is initiated
-	VolumeSnapshotRestoreStatusInitial SnapshotRestoreStatusType = ""
+	VolumeSnapshotRestoreStatusInitial VolumeSnapshotRestoreStatusType = ""
 	// VolumeSnapshotRestoreStatusPending for when restore is in pending state
-	VolumeSnapshotRestoreStatusPending SnapshotRestoreStatusType = "Pending"
+	VolumeSnapshotRestoreStatusPending VolumeSnapshotRestoreStatusType = "Pending"
 	// VolumeSnapshotRestoreStatusRestore for when restore is in restore state
-	VolumeSnapshotRestoreStatusRestore SnapshotRestoreStatusType = "Restore"
+	VolumeSnapshotRestoreStatusRestore VolumeSnapshotRestoreStatusType = "Restore"
```
Comment (disrani-px) [inline pkg/apis/stork/v1alpha1/snapshotrestore.go]: This state is confusing. `Restore` implies it is in restore process. Maybe rename to `Restored`?

Comment (ram-infrac) [reply pkg/apis/stork/v1alpha1/snapshotrestore.go]: restore stage is when snapshot restore objects are ready eg. downloaded

**Pull request merged**